src/PowerShellEditorServices/Workspace/Workspace.cs

```diff
@@ -202,6 +213,35 @@ public string GetRelativePath(string filePath)
             return resolvedPath;
         }
 
+        internal static string MapRemotePathToLocal(string remotePath, string connectionString)
+        {
+            string sessionTempPath =
+                Path.Combine(
+                    Path.GetTempPath(),
+                    "PSES-" + Process.GetCurrentProcess().Id,
+                    "RemoteFiles");
+
+            // The path generated by this code will look something like
+            // %TEMP%\PSES-[PID]\RemoteFiles\1205823508\computer-name\MyFile.ps1
```

Comment:
That's a good point, though the one thing I worry about is the locally cached file being out of date.  For example, the user connects to a remote machine to debug a script running there, finds a bug, fixes it and re-deploys the script.  They attempt to debug the same script again remotely but since we already have the cached file we show the old version.

Maybe I could send the local copy's timestamp to the server to determine whether we need to sync, but that might be a little overkill.  I'm certainly interested to go that direction in the future if the need arises.

I guess this problem will exist with the current approach if PSES somehow ends up with the same PID after X launches, or even if they change the remote file while in the same editing session.  Maybe I should proactively delete that remote files folder once the remote runspace session ends?